superset/db_engine_specs/superset.py

```diff
@@ -0,0 +1,330 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import datetime
+import operator
+import urllib.parse
+from functools import wraps
+from typing import (
+    Any,
+    Callable,
+    cast,
+    Dict,
+    Iterator,
+    List,
+    Optional,
+    Tuple,
+    Type,
+    TypeVar,
+)
+
+from flask import g
+from flask_login import current_user
+from shillelagh.adapters.base import Adapter
+from shillelagh.backends.apsw.dialect import APSWDialect
+from shillelagh.exceptions import ProgrammingError
+from shillelagh.fields import (
+    Blob,
+    Boolean,
+    Date,
+    DateTime,
+    Field,
+    Float,
+    Integer,
+    Order,
+    String,
+    Time,
+)
+from shillelagh.filters import Equal, Filter, Range
+from shillelagh.types import RequestedOrder, Row
+from sqlalchemy import MetaData, Table
+from sqlalchemy.engine.url import URL
+from sqlalchemy.exc import NoSuchTableError
+from sqlalchemy.pool.base import _ConnectionFairy
+from sqlalchemy.sql import Select, select
+
+from superset import db, security_manager, sql_parse
+from superset.db_engine_specs.sqlite import SqliteEngineSpec
+
+
+class SupersetEngineSpec(SqliteEngineSpec):
+    """
+    Internal engine for Superset
+
+    This DB engine spec is a meta-database. It uses the shillelagh library
+    to build a DB that can operate across different Superset databases.
+    """
+
+    engine = "superset"
+    engine_name = "Superset"
+
+
+# pylint: disable=abstract-method
+class SupersetAPSWDialect(APSWDialect):
+
+    """
+    A SQLAlchemy dialect for an internal Superset engine.
+
+    This dialect allows query to be executed across different Superset
```

Comment:
`This dialect allows queries to be executed across different Superset` better?

Reply:
Better!